src/socket_io/emitter.cr

```diff
@@ -0,0 +1,51 @@
+require "./event"
+require "json"
+
+module SocketIO
+  alias EventHandler = Event ->
+
+  class Emitter
+    @listeners : Hash(String, Array(EventHandler)) = Hash(String, Array(EventHandler)).new
```

Comment:
Why use an unsafe object and then sync it with mutex instead of using GenericActor shard or better yet a channel?